components/post_view/combined_system_message/combined_system_message.jsx

```diff
@@ -176,34 +186,48 @@ export default class CombinedSystemMessage extends React.PureComponent {
         const {
             actions,
             allUserIds,
+            allUsernames,
         } = this.props;
-        const missingProfiles = await actions.getMissingProfilesByIds(allUserIds);
-        this.setState({missingProfiles});
+        let missingProfilesOnIds = [];
+        if (allUserIds.length > 0) {
+            missingProfilesOnIds = await actions.getMissingProfilesByIds(allUserIds);
+        }
+
+        let missingProfilesOnUsernames = [];
+        if (allUsernames.length > 0) {
+            missingProfilesOnUsernames = await actions.getMissingProfilesByUsernames(allUsernames);
+        }
+
+        this.setState({
+            missingProfilesOnIds,
+            missingProfilesOnUsernames,
+        });
```

Comment:
Is setting the state needed? I don't see these being used anywhere

Reply:
Yes, it's a quick hack to re-render component whenever there are missing profiles.

Reply:
After getting missing profiles, those are loaded from the `UserStore` - [ref](https://github.com/mattermost/mattermost-webapp/blob/master/utils/utils.jsx#L1144) - (which I should have not used in the first place) and since props do not change, the component does not re-render.  The state changes does the trick.

Anyway, I realized that the logic is not really explicit and also the component should be more of redux-style implementation. With that, I've updated the logic following the implementation I've done on mobile app.